### PR TITLE
recurse findOptions on non root package.json

### PR DESCRIPTION
**Pull request by zetaron**
# Why the hassle?
For a recent project I was in need to not use the first `package.json` found to configure the `standard-engine` but walk up the tree into the projects root, where the actual linter configuration is happening.

# What did I do to achive the goal?
To achive this I extended the `findOptions` method to check for the special option `isProjectRoot` to be `false` and if so walk up the file tree to find an other `package.json`.

I've added a test and a fixture folder to demonstrate it working properly.

### CODE REVIEW

**Comment by novemberborn** [comment]
Interesting!

I think this should go into the `standard-engine` field instead:

```json
{
  "standard-engine": {
    "skipPackageJson": true
  }
}
```

Or something along those lines.

**Comment by zetaron** [comment]
@novemberborn: That sounds better that smapping around in the global namesapce, I'll update the code.

**Comment by zetaron** [comment]
@novemberborn Sadly this change would conflict with the way `standard-engine` is currently used, and would result in a breaking change. 

I could imagine using the `standard` configuration object instead of the `standard-engine` as it is already used as an object to host multiple other settings.

**Comment by gustavnikolaj** [comment]
We can update the standard-engine field to be an object instead of a string and convert a string in that field to an object with a linterName property before to handle backwards compatibility and avoiding the breaking change. 

I don't like invading other people's namespaces ;-)

**Comment by zetaron** [comment]
That's an option I didn't think of.
I'll update the commits.

**Comment by zetaron** [comment]
This commit brings the backwards compatibility and searching for the `skipPackageJson` inside the `standard-engine` configuration scope.

**Comment by zetaron** [comment]
Seems like the following test is at fault.
It also seems like it's normal for this test to be at around ~1200ms on both Travis and AppVeyor (Prepare 2.0.0 release: ee939f1) it has been higher before, so I assume this is some sort of CI related timing issue.

Travis:
```
linter-js-standard-engine
    1) should be able to lint a test file

1) linter-js-standard-engine should be able to lint a test file:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```
AppVeyor:
```
linter-js-standard-engine
    √ should be able to lint a test file (1438ms)
```

File: lib/findOptions.js [hunk 1]
```diff
@@ -49,28 +49,42 @@ function getOptions (packagePath) {
...
-            linters.push(dep)
-          }
+      // Parse standard-engine configuration
+      if (pkg['standard-engine']) {
+        if (typeof pkg['standard-engine'] === 'string') {
+          standardEngineConfiguration.linterName = pkg['standard-engine']
         }
+
+        const isSkip = pkg['standard-engine'].skipPackageJson
+        standardEngineConfiguration.skipPackageJson = (isSkip || false)
```
Comment (novemberborn) [inline lib/findOptions.js]: I think this is a bit more to the point:

```js
let opts = pkg['standard-engine']
if (typeof opts === 'string') {
  opts = {linterName: opts, skipPackageJson: false}
} else {
  opts.skipPackageJson = opts.skipPackageJson === true
}
```

**Review by novemberborn — COMMENTED** [review]
> Seems like the following test is at fault.
> It also seems like it's normal for this test to be at around ~1200ms on both Travis and AppVeyor (Prepare 2.0.0 release: ee939f1) it has been higher before, so I assume this is some sort of CI related timing issue.

I've restarted the CI run. @gustavnikolaj perhaps we need to increase timing allowances?

I've got a small nit around the `pkg['standard-engine']` parsing, otherwise this is good to go assuming CI passes. Nice work @zetaron!

File: lib/findOptions.js (hunk 1 above)
Comment (zetaron) [reply lib/findOptions.js]: I've added an other commit to incorporate your suggestion.

**Comment by novemberborn** [comment]
@zetaron something still didn't sit right with me. Rather than putting you through another back-and-forth I figured I'd try to resolve my discomfort myself. See the commit I just pushed, let me know what you think.

If you're happy then let's wait a bit for @gustavnikolaj to chime in, and then I'll land this. Many many thanks for your hard work on this 🥇 .

**Comment by zetaron** [comment]
Looks good to me. 

Also no offense taken, that's the reason I did leave the option active for contributors to contribute to the PR :).

**Review by gustavnikolaj — APPROVED** [review]
LGTM 👍

I really like how this came together :-)

**Comment by gustavnikolaj** [comment]
Merged and released in 2.1.0 :-)

Thanks for the great work :trophy:

**Pull request merged**